### ISSUE TITLE
Compile warnings when using withSpy, withMock

### ISSUE THREAD

**Issue by yaskor** [dinject/dinject#24]
Im getting compile warnings when Im using follwing construct....

```java		
private final List<Class> withMocks;
private final List<Class> withSpies;

private ContextExtension(List<Class> withMocks, List<Class> withSpies) {
	this.withMocks = withMocks;
	this.withSpies = withSpies;
}
		
...
		
BootContext bootContext = new BootContext();
withMocks.forEach(mock -> bootContext.withMock(mock));
withSpies.forEach(spy -> bootContext.withSpy(spy));
beanContext = bootContext.load();
```

```
Changes detected - recompiling the module!
Compiling 26 source files to C:\Users\saka\Documents\NetBeansProjects\private\javalin-sample\target\test-classes
de/yaskor/autocontract/test/ContextExtension.java:[68,55] unchecked method invocation: method withMock in class io.dinject.BootContext is applied to given types
  required: java.lang.Class<D>
  found: java.lang.Class
de/yaskor/autocontract/test/ContextExtension.java:[68,56] unchecked conversion
  required: java.lang.Class<D>
  found:    java.lang.Class
de/yaskor/autocontract/test/ContextExtension.java:[68,26] unchecked method invocation: method forEach in class java.util.ArrayList is applied to given types
  required: java.util.function.Consumer<? super E>
  found: java.util.function.Consumer<java.lang.Class>
de/yaskor/autocontract/test/ContextExtension.java:[69,53] unchecked method invocation: method withSpy in class io.dinject.BootContext is applied to given types
  required: java.lang.Class<D>
  found: java.lang.Class
de/yaskor/autocontract/test/ContextExtension.java:[69,54] unchecked conversion
  required: java.lang.Class<D>
  found:    java.lang.Class
de/yaskor/autocontract/test/ContextExtension.java:[69,26] unchecked method invocation: method forEach in class java.util.ArrayList is applied to given types
  required: java.util.function.Consumer<? super E>
  found: java.util.function.Consumer<java.lang.Class>
```

How can I satisfy ```Class<D>```?

**Comment by rbygrave** [comment]
Change it adding `<?>` to become:

```java
private final List<Class<?>> withMocks;
private final List<Class<?>> withSpies;
```
On the other hand, those methods that don't take a consumer can themselves change to take `Class<?>` so I may as well make that change - then use of raw `Class` is deemed ok (so one less thing for people to think about).

**Comment by rbygrave** [comment]
Ok, pushed that change - will consider this closed in 1.7
